integrations/pytorch/tutorials/woodfisher_pruning.md

```diff
@@ -0,0 +1,126 @@
+
+```yaml
+pruning_modifiers:
+
+  - !MFACPruningModifier
+    params: __ALL_PRUNABLE__
+    init_sparsity: 0.05
+    final_sparsity: 0.85
+    start_epoch: 1.0
+    end_epoch: 61.0
+    update_frequency: 4.0
+    mfac_options:
+      num_grads: {0.0: 256, 0.5: 512, 0.75: 1024, 0.83: 1400}
+      fisher_block_size: 10000
+      available_gpus: ["cuda:0"]
+```
+
+## WoodFisher Parameters
+The following parameters can be specified under the `mfac_options` parameter to control
+how the WoodFisher calculations are made. Ideal values will depend on the system
+available to run on and model to be pruned.
+
+### num_grads
+To approximate the second order information in the WoodFisher algorithm, first order
+gradients are used. `num_grads` specifies the number of recent gradient samples to store
+of a model while training.
+
+This value can be an int where that constant value will be used throughout pruning or a
+dictionary of float sparsity values to the number of gradients that should be
+stored when that sparsity level (between 0.0 and 1.0) is reached. If a
+dictionary, then 0.0 must be included as a key for the base number of gradients
+to store (i.e. {0: 64, 0.5: 128, 0.75: 256}).
+
+Storing gradients can be expensive, as for a dense model, each additional gradient
+sample stored requires about the same memory that the entire model needs. This is why
+the dictionary option allows for more gradients to be stored as the model gets more
+sparse.
+
+If a WoodFisher pruning run is unexpectedly killed, the reason could likely be that
+the gradient storage requirements exceeded the system's RAM. A safe rule of thumb for
+initial number of gradients is the number should be no greater than 1/4 of the
+available CPU RAM divided by the model size.
+
+
+### fisher_block_size
+To limit the computational cost of calculating second order information, the WoodFisher
+algorithm may compute a block diagonal matrix of a certain block size that is
+sufficient for generating the necessary information for pruning.
+
+The `fisher_block_size` specifies this block size.  If using GPUs to perform the
+WoodFisher computations, the GPUs should have `num_grads * fisher_block_size` extra
+memory duing training so each block can be stored and computed sequentially on a GPU.
+
+The default block size is 2000, and generally block sizes between 1000 and 10000 may be
+ideal. If `None` is provided, the full matrix will be computed without blocks.
+
+
+### available_gpus
+`available_gpus` is a list of GPU devices names to perform the WoodFisher computation
+with. If not provided, computation will be done on the CPU.
+
+
+## Wrap-Up
+This tutorial shows how to use WoodFisher pruning by swapping in a single modifier
+to a SparseML pruning recipe. Everything else for pruning and exporting stays the same.
+More papers and results will be posted as they become available.
+
+Open an
+[issue](https://github.com/neuralmagic/sparseml/issues) with any questions or bugs.
```

Comment:
```suggestion
[issue](https://github.com/neuralmagic/sparseml/issues) with any bugs or feature requests.

```suggestion

For Neural Magic Support, sign up or log in to get help with your questions in our **Tutorials channel:** [Discourse Forum](https://discuss.neuralmagic.com/) and/or [Slack](https://join.slack.com/t/discuss-neuralmagic/shared_invite/zt-q1a1cnvo-YBoICSIw3L1dmQpjBeDurQ). 

```
```